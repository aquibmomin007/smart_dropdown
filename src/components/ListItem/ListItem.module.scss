$lightGrey: #ccc;
$lightGreyOpacity: #cccccc4d;
$activeGrey: #d8d8d8;

.dropdownListItem {
    padding: 5px 10px;
    border-bottom: 1px solid $lightGrey;
    cursor: pointer;
    word-break: break-word;

    &:hover{
        background: $lightGreyOpacity
    }
}

.activeListItem {
    background: $activeGrey;
}