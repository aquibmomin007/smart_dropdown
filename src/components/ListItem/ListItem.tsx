import React, {  } from 'react';
import cx from 'classnames';
import styles from './ListItem.module.scss'
import { SelectOption } from '../../hooks/useCountriesList';

export type ListItemProps = {
    selectedOption: SelectOption | null;
    country: SelectOption,
    handleSelectChange: (country: SelectOption) => () => void
}

const ListItem = ({ selectedOption, country, handleSelectChange }: ListItemProps) => { 
  return (
    <li
        data-testid="list-item-test"
        className={cx(styles.dropdownListItem, {
            [styles.activeListItem]: selectedOption && selectedOption.value === country.value
        })}
        onClick={handleSelectChange(country)}>
            {country.label}
    </li>
  )
}

export default ListItem;