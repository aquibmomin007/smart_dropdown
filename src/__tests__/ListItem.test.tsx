import React from 'react';
import ListItem, {ListItemProps} from '../components/ListItem/ListItem';
import { render } from '@testing-library/react';


describe('ListItem Testing', () => {
  test('renders correctly for nothing is selected', () => {
    const listItemProps:ListItemProps = {
      selectedOption: null,
      country: {
        label: "Singapore",
        value: 'SG'
      },
      handleSelectChange: jest.fn()
    };

    const { container, getByTestId } = render(
        <ListItem {...listItemProps} />
    )
    const listItemHtml = getByTestId('list-item-test');
    expect(container).toMatchSnapshot();
    expect(listItemHtml.classList.contains('activeListItem')).not.toBe(true)
  });

  test('renders correctly for Item is selected', () => {
    const selectedItem = {
      label: "Singapore",
      value: 'SG'
    };
    const listItemProps:ListItemProps = {
      selectedOption: selectedItem,
      country: selectedItem,
      handleSelectChange: jest.fn()
    };

    const { container, getByTestId } = render(
        <ListItem {...listItemProps} />
    )

    const listItemHtml = getByTestId('list-item-test');
    expect(container).toMatchSnapshot();
    expect(listItemHtml.classList.contains('activeListItem')).toBe(true)

  });
});
