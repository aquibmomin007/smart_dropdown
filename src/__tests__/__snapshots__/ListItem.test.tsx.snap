// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Testing renders correctly for Item is selected 1`] = `
<div>
  <li
    class="dropdownListItem activeListItem"
    data-testid="list-item-test"
  >
    Singapore
  </li>
</div>
`;

exports[`ListItem Testing renders correctly for nothing is selected 1`] = `
<div>
  <li
    class="dropdownListItem"
    data-testid="list-item-test"
  >
    Singapore
  </li>
</div>
`;
